File: June Challenge/Week 1/C++/DAY_1/InvertBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */


void invert(TreeNode* node){

    if(node==NULL){
        return;
    }
    TreeNode *temp;
    invert(node->left);
    invert(node->right);

    temp=node->left;
    node->left=node->right;
    node->right=temp;

}

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        invert(root);
        return  root;
    }
};

File: June Challenge/Week 1/Java/Day_1/Invert_Binary_Tree.java
//Problem:https://leetcode.com/problems/invert-binary-tree/
//solution:
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
class Solution {
    public TreeNode invertTree(TreeNode root) {
        Invert(root);
        return root;
    }
    public static void Invert(TreeNode node){
        if(node==null){
        return;
        }
        if ((node.left==null)&&(node.right==null)){
            return;
        }
        TreeNode newnode=new TreeNode();
        newnode=node.left;
        node.left=node.right;
        node.right=newnode;
        Invert(node.left);
        Invert(node.right);
        
    }
}
